src/ripple/core/tests/Config.test.cpp

```diff
@@ -346,10 +441,160 @@ port_wss_admin
+n9L81uNCaPgtUJfaHh89gmdvXKAmSt5Gdsw2g1iPWaPkAHW5Nm4C
+
+[validator_keys]
+nHUhG1PgAG8H8myUENypM35JgfqXAKNQvRVVAFDRzJrny5eZN8d5
+nHBu9PTL9dn2GuZtdW4U2WzBwffyX9qsQCd9CNU4Z5YG3PQfViM8
+
+[validation_quorum]
+4
+)rippleConfig");
+            c.loadFromString (toLoad);
+            expect (c.legacy ("validators_file").empty ());
+            expect (c.section (SECTION_VALIDATORS).values ().size () == 3);
+            expect (c.section (SECTION_VALIDATOR_KEYS).values ().size () == 2);
+            expect (c.VALIDATION_QUORUM == 4);
+        }
+        {
+            // load from specified [validators_file]
+            int const quorum = 3;
+            detail::ValidatorsTxtGuard vtg (
+                "test_cfg", "validators.cfg", quorum);
+            expect (vtg.validatorsFileExists ());
+            Config c;
+            boost::format cc ("[validators_file]\n%1%\n");
+            c.loadFromString (boost::str (cc % vtg.validatorsFile ()));
+            expect (c.legacy ("validators_file") == vtg.validatorsFile ());
+            expect (c.section (SECTION_VALIDATORS).values ().size () == 5);
+            expect (c.section (SECTION_VALIDATOR_KEYS).values ().size () == 3);
+            expect (c.VALIDATION_QUORUM == quorum);
+        }
+        {
+            // load from validators file in default location
+            int const quorum = 3;
+            detail::ValidatorsTxtGuard vtg (
+                "test_cfg", "validators.txt", quorum);
+            detail::RippledCfgGuard rcg ("test_cfg", "", "");
+            expect (vtg.validatorsFileExists ());
+            expect (rcg.configFileExists ());
+            auto& c (rcg.config ());
+            expect (c.legacy ("validators_file").empty ());
+            expect (c.section (SECTION_VALIDATORS).values ().size () == 5);
+            expect (c.section (SECTION_VALIDATOR_KEYS).values ().size () == 3);
+            expect (c.VALIDATION_QUORUM == quorum);
+        }
+        {
+            // load from specified [validators_file] instead
+            // of default location
+            int const quorum = 3;
+            detail::ValidatorsTxtGuard vtg (
+                "test_cfg", "validators.cfg", quorum);
+            expect (vtg.validatorsFileExists ());
+            detail::ValidatorsTxtGuard vtgDefault (
+                "test_cfg", "validators.txt", 4);
+            expect (vtgDefault.validatorsFileExists ());
+            detail::RippledCfgGuard rcg (
+                "test_cfg", "", vtg.validatorsFile ());
+            expect (rcg.configFileExists ());
+            auto& c (rcg.config ());
+            expect (c.legacy ("validators_file") == vtg.validatorsFile ());
+            expect (c.section (SECTION_VALIDATORS).values ().size () == 5);
+            expect (c.section (SECTION_VALIDATOR_KEYS).values ().size () == 3);
+            expect (c.VALIDATION_QUORUM == quorum);
+        }
+        {
+            // do not load quorum from validators file if in config
```

Comment:
how about:
`do not load quorum from validators file if in rippled config file?`